torchgen/model.py

```diff
@@ -146,6 +177,24 @@ def parse(value: str) -> "DispatchKey":
         raise AssertionError(f"unknown dispatch key {value}")
 
 
+def codegen_per_backend_entries() -> str:
+    r = []
+    for fk in FUNCTIONALITY_KEYS:
+        for bc in BACKEND_COMPONENTS:
+            r.append(f"    {fk}{bc} = auto()")
+    return "\n".join(r)
+
+
+for fk in FUNCTIONALITY_KEYS:
+    for bc in BACKEND_COMPONENTS:
+        if not hasattr(DispatchKey, fk + bc):
+            r = codegen_per_backend_entries()
```

Comment:
I wonder if we may as well just use this to generate this list directly, instead of being forced to manually write it out and use this to assert that it's right.

I'm not too opinionated - manually writing it all out makes it easier for people staring at the codegen to know what the actual dispatch keys are anyway.

Reply:
The main reason I manually generated was I couldn't figure out how to make mypy work correctly with autogen.